phpBB/assets/javascript/core.js

```diff
@@ -263,7 +263,11 @@ phpbb.ajaxify = function(options) {
 			}
 			phpbb.clearLoadingTimeout();
 			var errorText = false;
-			if (typeof errorThrown === 'string' && errorThrown.length > 0) {
+			var responseText = $.parseJSON(jqXHR.responseText);
```

Comment:
Yep, if it's an error handler it's pretty likely that this isn't JSON at all (for example, a 500 error). It should be wrapped in a try-catch block, and `JSON.parse` should be used, not `$.parseJSON`—phpBB doesn't support IE7!